pkg/segment/reader/record/recordreader.go

```diff
@@ -36,6 +36,171 @@ import (
 	log "github.com/sirupsen/logrus"
 )
 
+func ReadAllColsForRRCs(segKey string, vTable string, rrcs []*utils.RecordResultContainer,
```

Comment:
if this toplevel function gets called multiple times on multiple segkeys in during one query run time. Then we should think if passing in a array of array of RRCs, so that we can reuse the rrc array. The cval enclosures are small sizes but when done on thousands of records/columns, they can create GC pressure. But first we should calculate out how many times this function is going to get called on a query and then decide if we want to pre-alloc. 
This does not need to be done now. Just making a note of this as an optimization we can think of when we start doing perf tests.

Reply:
Yes, passing in an array of CValEnclosures (I think that's what you meant?) could be a good optimization. I'll wait for benchmarking first though, and not do that in this PR